### PR TITLE
CP-17647: Implement PVS_server.[introduce|forget]

### PR DESCRIPTION
**Pull request by johnelse**

### CODE REVIEW

File: ocaml/xapi/xapi_pvs_server.ml [hunk 1]
```diff
@@ -14,13 +14,38 @@
...
 
+open Listext
+
 module D = Debug.Make(struct let name = "xapi_pvs_server" end)
 
-let not_implemented x =
-	raise (Api_errors.Server_error (Api_errors.not_implemented, [ x ]))
+let assert_ports_valid ~first_port ~last_port =
+	let assert_port_valid ~port ~name =
+		if port < 0L || port > 65535L
```
Comment (robhoes) [inline ocaml/xapi/xapi_pvs_server.ml]: Port 0 should not be allowed.

File: ocaml/xapi/xapi_pvs_server.ml [hunk 2]
```diff
@@ -14,13 +14,38 @@
 
 (** Implementation of PVS Server *)
 
+open Listext
+
 module D = Debug.Make(struct let name = "xapi_pvs_server" end)
 
-let not_implemented x =
-	raise (Api_errors.Server_error (Api_errors.not_implemented, [ x ]))
+let assert_ports_valid ~first_port ~last_port =
```
Comment (robhoes) [inline ocaml/xapi/xapi_pvs_server.ml]: This looks like a generally useful function that could go in helpers.ml alongside `assert_is_valid_ip`. Perhaps call it `assert_tcp_udp_port_range_valid`. Also `assert_port_valid` could be a top-level function in helpers.

File: ocaml/xapi/xapi_pvs_server.ml (hunk 1 above)
Comment (johnelse) [inline ocaml/xapi/xapi_pvs_server.ml]: Ok

File: ocaml/xapi/xapi_pvs_server.ml (hunk 2 above)
Comment (johnelse) [inline ocaml/xapi/xapi_pvs_server.ml]: Agreed, I'll pull it out.

File: ocaml/test/test_helpers.ml [hunk 3]
```diff
@@ -201,6 +288,8 @@ let test =
 	"test_helpers" >:::
 		[
 			"test_determine_gateway" >::: DetermineGateway.tests;
+			"test_assert_is_valid_ip" >::: PortCheckers.tests;
```
Comment (robhoes) [inline ocaml/test/test_helpers.ml]: Hmm... copy/paste error?

Comment (johnelse) [inline ocaml/test/test_helpers.ml]: oops...yes

**Comment by johnelse** [comment]
Fixed now.

**Comment by robhoes** [comment]
Thanks: :shipit:

**Pull request merged**